tests/cephadm/test_client.py

```diff
@@ -43,35 +45,56 @@ def put_file(client, file_name, content, file_mode, sudo=True):
         file_.write(content)
         file_.flush()
 
-    if config.get("node"):
-        node = get_node_by_id(cls.cluster, config["node"])
-
-        # Copy the keyring to client
-        node.exec_command(sudo=True, cmd="mkdir -p /etc/ceph")
-        put_file(node, client_file, cnt_key, "w")
-
-        if config.get("copy_ceph_conf", True):
-            # Get minimal ceph.conf
-            ceph_conf, err = cls.shell(args=["ceph", "config", "generate-minimal-conf"])
-            # Copy the ceph.conf to client
-            put_file(node, "/etc/ceph/ceph.conf", ceph_conf, "w")
-
-        # Copy admin keyring to client node
-        if config.get("copy_admin_keyring"):
-            admin_keyring, _ = cls.shell(args=["ceph", "auth", "get", "client.admin"])
-            put_file(node, "/etc/ceph/ceph.client.admin.keyring", admin_keyring, "w")
-
-        # Install ceph-common
-        if config.get("install_packages"):
-            for pkg in config.get("install_packages"):
-                node.exec_command(cmd=f"yum install -y --nogpgcheck {pkg}", sudo=True)
-
-        out, _ = node.exec_command(cmd="ls -ltrh /etc/ceph/", sudo=True)
-        log.info(out)
-
-    # Hold local copy of the client key-ring in the installer node
-    if config.get("store-keyring"):
-        put_file(cls.installer, client_file, cnt_key, "w")
+    if config.get("nodes"):
+        nodes = config["nodes"]
+        if not isinstance(nodes, list):
+            nodes = [nodes]
+
+        def setup(host):
```

Comment:
The nested method is big. Could you please add a separate method

Reply:
Making it a separate method would mean we would have to pass too many params from the  `  put_file() ` method, making it complicated. The contents of that method was put into new nested method as we wanted to invoke parallel execution for client setup. 

Would add a new new method if absolutely necessary.